src/java/org/apache/cassandra/db/guardrails/GuardrailsConfigService.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.config.GuardrailsOptions;
+import org.apache.cassandra.utils.MBeanWrapper;
+
+/**
+ * Implementation of {@link GuardrailsConfigServiceMBean}.
+ */
+public class GuardrailsConfigService implements GuardrailsConfigServiceMBean
+{
+    public static final String MBEAN_NAME = "org.apache.cassandra.db:type=Guardrails";
+
+    private final GuardrailsOptions config = DatabaseDescriptor.getGuardrailsConfig();
```

Comment:
this works for default but feels meaningless for the plugin case as `GuardrailsConfigProvider` isn't used.

Now, we talked about the plugin case, so I am cool with this, but just calling out as I know that is a thing for your team